src/components/input/test/Select-test.js

```diff
@@ -425,12 +436,137 @@ describe('Select', () => {
     });
   });
 
+  context('when listbox does not fit in available space', () => {
+    async function getOpenListbox(wrapper) {
+      toggleListbox(wrapper);
+      await waitFor(() => !isListboxClosed(wrapper));
+
+      return getListbox(wrapper);
+    }
+
+    it('never renders a listbox bigger than the viewport', async () => {
+      const name = 'name'.repeat(1000);
+      const wrapper = createComponent(
+        { buttonContent: 'Select a value' },
+        {
+          items: ['1', '2', '3'].map(id => ({ id, name })),
+        },
+      );
+
+      const listbox = await getOpenListbox(wrapper);
+      const { width: listboxWidth } = listbox
+        .getDOMNode()
+        .getBoundingClientRect();
+
+      assert.isTrue(listboxWidth < window.window.innerWidth);
+    });
+
+    [
+      { name: 'long name'.repeat(50), shouldGrow: true },
```

Comment:
This string is about 2300px wide on my system (using `CanvasRenderingContext2D.measureText`). If the viewport happens to be really large in the test browser, it is possible the listbox might not need to grow.